### ISSUE TITLE
ListView with ScrollController throw Exception (You need to put controller before)

### ISSUE THREAD

**Issue by ghprod** [jonataslaw/getx#390]
**Describe the bug**
SecondScreen has ListView inside StreamBuilder with ScrollController. When going back via `Get.back` to HomeScreen, exception threw.

```
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following message was thrown building StreamBuilder<List<int>>(dirty, state: _StreamBuilderBaseState<List<int>, AsyncSnapshot<List<int>>>#70eaf):
 SecondController not found. You need call put<SecondController>(SecondController()) before

The relevant error-causing widget was: 
  StreamBuilder<List<int>> file:///home/bolt/dev/mobile/getx_debug/lib/main.dart:108:13
When the exception was thrown, this was the stack: 
#0      GetInstance.find (package:get/src/instance/get_instance.dart:162:9)
#1      GetView.controller (package:get/src/state/get_view.dart:14:37)
#2      SecondScreen.build.<anonymous closure> (package:getx_debug/main.dart:116:31)
#3      StreamBuilder.build (package:flutter/src/widgets/async.dart:509:81)
#4      _StreamBuilderBaseState.build (package:flutter/src/widgets/async.dart:127:48)
```

**To Reproduce**
1. Go to HomeScreen
2. Click on `Go to second`
3. Wait for seconds
4. Tap back icon/hardware
5. Show red box contains exception before going back to HomeScreen

**Expected behavior**
I've already close ScrollController via `onClose`,  but no luck. My expectation, the exception is gone.

**Screenshots**
https://dump.video/i/n1EZQk.mp4

**Flutter Version:**
Flutter 1.17.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8af6b2f038 (2 weeks ago) • 2020-06-30 12:53:55 -0700
Engine • revision ee76268252
Tools • Dart 2.8.4

**Get Version:**
3.3.0

**Describe on which device you found the bug:**
Android Emulator with API 23 Android 6.0

**Minimal reproduce code**

```dart
import 'dart:async';
import 'dart:math';

import 'package:flutter/material.dart';
import 'package:get/get.dart';

void main() => runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
//      initialBinding: HomeBinding(),
      initialRoute: '/',
      getPages: [
        GetPage(name: '/', page: () => HomeScreen(), binding: HomeBinding()),
        GetPage(
            name: '/second',
            page: () => SecondScreen(),
            binding: SecondBinding()),
      ],
    );
  }
}

class HomeBinding implements Bindings {
  @override
  void dependencies() {
    Get.lazyPut<HomeController>(() => HomeController());
  }
}

class HomeController extends GetxController {}

class HomeScreen extends GetView<HomeController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: FlatButton(
          onPressed: () {
            Get.toNamed('/second');
          },
          child: Text('Go to second'),
        ),
      ),
    );
  }
}

class SecondBinding implements Bindings {
  @override
  void dependencies() {
    Get.lazyPut<SecondController>(() => SecondController());
  }
}

class SecondController extends GetxController {
  Timer timer;

  ScrollController scrollController = ScrollController(
    initialScrollOffset: 0.0,
    keepScrollOffset: true,
  );

  StreamController<List<int>> controller = StreamController<List<int>>.broadcast();

  Stream<List<int>> get stream => controller.stream;

  @override
  void onInit() {
    super.onInit();

    scrollController.addListener(() {
      double offset = 0.9 * scrollController.position.maxScrollExtent;

      if (scrollController.position.pixels > offset) {
        print('scrollController offset > load more');
      }
    });

    timer = Timer.periodic(Duration(seconds: 1), (_) {
      print('Timer.periodic ${Random().nextInt(99)}');

      controller.add([
        Random().nextInt(99),
        Random().nextInt(9999),
        Random().nextInt(99999),
      ]);
    });
  }

  @override
  void onClose() {
    timer.cancel();
    scrollController.dispose();
    controller.close();

    super.onClose();
  }
}

class SecondScreen extends GetView<SecondController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: StreamBuilder(
          stream: controller.stream,
          builder: (context, snapshot) {
            if (snapshot.hasData) {
              final List<int> items = snapshot.data;

              if (items.isNotEmpty) {
                return ListView.separated(
                  controller: controller.scrollController,
                  separatorBuilder: (context, index) => Divider(height: 0.5),
                  itemCount: items.length,
                  itemBuilder: (context, index) {
                    int item = items[index];

                    return InkWell(
                      child: ListTile(
                        title: Text(item.toString()),
                      ),
                    );
                  },
                );
              } else {
                return Center(child: Text('Nothing here'));
              }
            } else if (snapshot.hasError) {
              return Center(child: Text('Something went wrong'));
            }

            return Center(child: CircularProgressIndicator());
          }),
    );
  }
}
```

**Comment by jonataslaw** [comment]
Thank you for opening this issue with a reproduction code.
I will take a look at this today after my shift.

**Comment by jonataslaw** [comment]
Fix on lastest update

**Comment by ghprod** [comment]
Thanks @jonataslaw

**Comment by ghprod** [comment]
Confirmed it's fixed with the latest version 3.4.0.
